### PR TITLE
Expands circuits with new and terrifying possibilities

### PR DESCRIPTION
**Pull request by Kubisopplay**
## About The Pull Request
Adds an extension to BCI, called Body Control Core. It is a modular system that allows for circuit controlled body operation, that may range from simple heals, to much more advanced components. Entire system is balanced by having a mechanic that I called stress. The more you use the modules, the more severe negatives the BCC will inflict on you. Since the modules will cause heavy debuffs they will need to be managed by the circuit controlling it, which should lead to some interesting development. To encourage interdep interaction, main ways of managing that stress will be items that can be made by cooks/bartenders, or chemists. Also, since stacking it with chemistry would encourage powergaming, the BCC will just by existing weaken chem effects.

## Why It's Good For The Game
Gives a potential for circuits to be a bit more impactful outside of admin events. Lets the science do a "mad science" projects, without taking over the place of medbay. Opens way for some antag only modules that should be interesting.

## Changelog

:cl:
add: Adds a circuit expansion module.

/:cl:

### CODE REVIEW

**Comment by san7890** [comment]
blocking per oranges on discord

File: code/modules/wiremod/shell/body_control_core.dm [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+	var/stress_cost = 0 /// amount of stress this module adds to the bcc when activated
+
+/obj/item/bcc_module/proc/trigger(delta_time, times_fired, mob)
+	return
+
+/obj/item/bcc_module/proc/stress(delta_time, mob, stress)
+	return
+
+
+/obj/item/bcc_module/basic_heal
```
Comment (Kylerace) [inline code/modules/wiremod/shell/body_control_core.dm:81]: we are not returning to buffing / nerfing nanites sorry

**Comment by Kubisopplay** [comment]
It was absolutely expected, but at least I can point fingers to this now.

**Comment by optimumtact** [comment]
please don't try to remake nanites, they have been removed, any similar style system will not get merged.

There are two existing systems that you can dabble with instead that have similar mechanics, chemistry and virology.

one of them is in need of a major overhaul.

**Comment by optimumtact** [comment]
also circuits that allow for combat related functionality such as heals or dealing damage, even to yourself, (ignoring signaller memes for now) are also things that will be close on sight

**Pull request closed without merging**